### DISCUSSION TITLE
Anvil assistance

### DISCUSSION THREAD

**Discussion by OnlyImmutable** [Minestom/Minestom#1440] [Help]
Hey guys,

I have written some code to generate an anvil gui, which works great it generates and shows up fine and everything works. 

The issue is, I want to type something in the Anvil text box, click the outputted name tag and grab the name and display it in chat.. except because the anvil generates the nametag its showing air?

```java
public final class QuestCreateNameInventory extends AnvilInventory {
    public QuestCreateNameInventory(Player player) {
        super(Component.text("Enter name..").color(NamedTextColor.GRAY));

        setItemStack(0, ItemStack.of(Material.NAME_TAG).withDisplayName(Component.text("").color(NamedTextColor.GRAY)));

        // Add slot conditions
        addInventoryCondition((clicker, slot, type, result) -> {
            if (slot == 0 || slot == 1) {
                clicker.sendMessage(Component.text("You must select the finished tag!").color(NamedTextColor.GRAY));
                result.setCancel(true);
                return;
            }

            AnvilInventory inventory = (AnvilInventory) clicker.getOpenInventory();

            if (inventory == null) {
                result.setCancel(true);
                return;
            }

            ItemStack item = inventory.getItemStack(2);

            clicker.sendMessage(item.toString());

            clicker.closeInventory();
            result.setCancel(true);
        });

        player.openInventory(this);
    }
}
```

I need to figure out why it won't detect the nametag generation, I assume packets have something to do with it?

You can see an example [here](https://i.gyazo.com/17fb065e322168a8465642861b69240b.mp4)

**Comment by KrystilizeNevaDies** [comment]
Minestom has no built in logic for this, you will need to add it yourself.

It's kind of on the fence about whether it belongs in minestom, on one hand it is client prediction, but on the other hand it is not really minestom's place to enforce that this specific prediction is done on the server.

**Reply by OnlyImmutable** [reply]
Do you have any guidance for how to add this logic to the server so I can track?

Or alternatively, how to stop the OpenSignEditor packet from closing every time it is opened?

**Comment by mworzala** [comment]
The client sends a [Rename Item](https://wiki.vg/Protocol#Rename_Item) packet when typing in the anvil GUI, the item name changing client side is prediction only. You will need to track this packet for the player with the GUI open to determine the text they wrote. You can do this by listening to the [`PlayerPacketEvent`](https://javadoc.minestom.net/net/minestom/server/event/player/PlayerPacketEvent.html).
